NotesBackEnd/src/Notes.Dependency/Containers/DependencyContainer.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Notes.Contracts.Dependency;
+using Notes.Dependency.LifeTimeManagers;
+using Unity;
+using Unity.Injection;
+
+namespace Notes.Dependency.Containers
+{
+    public class DependencyContainer : IDependencyContainerRegister, IDependencyContainerResolver
+    {
+        private readonly IUnityContainer _unityContainer;
+
+        public DependencyContainer()
+        {
+            _unityContainer = new UnityContainer();
+        }
+
+        private DependencyContainer(IUnityContainer container)
+        {
+            _unityContainer = container;
+        }
+
+        public IDependencyContainerRegister RegisterType<TTo>(Func<TTo> getObjectFunc, LifetimeTypes lifetimeType)
+        {
+            var lifetimeManager = lifetimeType.GetUnityLifetimeManager();
+
+            _unityContainer.RegisterType<TTo>(lifetimeManager, new InjectionFactory(_ => getObjectFunc()));
+            
+            return this;
+        }
+
+        public IDependencyContainerRegister RegisterType<TFrom, TTo>(LifetimeTypes lifetimeType) where TTo : TFrom
+        {
+            var lifetimeManager = lifetimeType.GetUnityLifetimeManager();
+
+            _unityContainer.RegisterType<TFrom, TTo>(lifetimeManager);
+
+            return this;
+        }
+
+        public object Resolve(Type serviceType)
+        {
+            try
+            {
+                return _unityContainer.Resolve(serviceType);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+
+        }
+
+        public IEnumerable<object> ResolveAll(Type serviceType)
+           => _unityContainer.ResolveAll(serviceType);

```

Comment:
Have a look at the implementation. This can actually throw the very same `Exception`s as the `_unityContainer.Resolve`…